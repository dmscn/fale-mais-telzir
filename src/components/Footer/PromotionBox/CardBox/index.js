export {default} from './CardBox';
export * from './CardBox';