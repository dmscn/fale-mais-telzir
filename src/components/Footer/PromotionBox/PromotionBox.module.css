.Container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  padding-bottom: 2em;
}

.Header {
  grid-area: header;
  display: flex;
  justify-content: center;
  color: white;
  margin-top: 2em;
}

.CardsContainer {
  display: flex;
  justify-content: center;
  align-items: center;
}
