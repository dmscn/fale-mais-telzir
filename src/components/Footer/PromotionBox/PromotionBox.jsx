// @flow
import React, { Component } from 'react'

import styles from './PromotionBox.module.css';
import plans from '../../../static/fale-mais-plans';

import CardBox from './CardBox';


export default class PromotionBox extends Component {
  render() {
    return (
      <section className={styles.Container}>
        <h3 className={styles.Header}>Escolha um de nossos planos</h3>

        <div className={styles.CardsContainer}>
          {plans.map(plan => (
            <CardBox
              destac={plan.name === 'FaleMais 60'}
              descount={plan.descountMin}
              title={plan.name}
              price={plan.descountMin}
            />
          ))}
        </div>
      </section>
    );
  }
}