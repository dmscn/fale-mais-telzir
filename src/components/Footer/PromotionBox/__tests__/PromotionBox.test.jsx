import React from 'react'
import PromotionBox from '../PromotionBox';

describe('<PromotionBox />', () => {
  it('should render', () => {
    const wrapper = shallow(<PromotionBox />);
    expect(wrapper).toMatchSnapshot();
  });
});