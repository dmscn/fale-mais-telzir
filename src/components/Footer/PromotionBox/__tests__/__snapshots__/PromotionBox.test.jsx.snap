// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromotionBox /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PromotionBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="Header"
        >
          Escolha um de nossos planos
        </h3>,
        <div
          className="CardsContainer"
        >
          <CardBox
            descount={30}
            destac={false}
            price={30}
            title="FaleMais 30"
          />
          <CardBox
            descount={60}
            destac={true}
            price={60}
            title="FaleMais 60"
          />
          <CardBox
            descount={120}
            destac={false}
            price={120}
            title="FaleMais 120"
          />
        </div>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Escolha um de nossos planos",
          "className": "Header",
        },
        "ref": null,
        "rendered": "Escolha um de nossos planos",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CardBox
              descount={30}
              destac={false}
              price={30}
              title="FaleMais 30"
            />,
            <CardBox
              descount={60}
              destac={true}
              price={60}
              title="FaleMais 60"
            />,
            <CardBox
              descount={120}
              destac={false}
              price={120}
              title="FaleMais 120"
            />,
          ],
          "className": "CardsContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "FaleMais 30",
            "nodeType": "class",
            "props": Object {
              "descount": 30,
              "destac": false,
              "price": 30,
              "title": "FaleMais 30",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "FaleMais 60",
            "nodeType": "class",
            "props": Object {
              "descount": 60,
              "destac": true,
              "price": 60,
              "title": "FaleMais 60",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "FaleMais 120",
            "nodeType": "class",
            "props": Object {
              "descount": 120,
              "destac": false,
              "price": 120,
              "title": "FaleMais 120",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="Header"
          >
            Escolha um de nossos planos
          </h3>,
          <div
            className="CardsContainer"
          >
            <CardBox
              descount={30}
              destac={false}
              price={30}
              title="FaleMais 30"
            />
            <CardBox
              descount={60}
              destac={true}
              price={60}
              title="FaleMais 60"
            />
            <CardBox
              descount={120}
              destac={false}
              price={120}
              title="FaleMais 120"
            />
          </div>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Escolha um de nossos planos",
            "className": "Header",
          },
          "ref": null,
          "rendered": "Escolha um de nossos planos",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CardBox
                descount={30}
                destac={false}
                price={30}
                title="FaleMais 30"
              />,
              <CardBox
                descount={60}
                destac={true}
                price={60}
                title="FaleMais 60"
              />,
              <CardBox
                descount={120}
                destac={false}
                price={120}
                title="FaleMais 120"
              />,
            ],
            "className": "CardsContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "FaleMais 30",
              "nodeType": "class",
              "props": Object {
                "descount": 30,
                "destac": false,
                "price": 30,
                "title": "FaleMais 30",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "FaleMais 60",
              "nodeType": "class",
              "props": Object {
                "descount": 60,
                "destac": true,
                "price": 60,
                "title": "FaleMais 60",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "FaleMais 120",
              "nodeType": "class",
              "props": Object {
                "descount": 120,
                "destac": false,
                "price": 120,
                "title": "FaleMais 120",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
